src/main/java/org/broadinstitute/hellbender/tools/copynumber/ModelSegments.java

```diff
@@ -0,0 +1,570 @@
+            final OverlapDetector<CopyRatio> copyRatioOverlapDetector = denoisedCopyRatios.getOverlapDetector();
+            filteredAllelicCounts = new AllelicCountCollection(
+                    sampleMetadata,
+                    filteredAllelicCounts.getRecords().stream()
+                            .filter(copyRatioOverlapDetector::overlapsAny)
+                            .collect(Collectors.toList()));
+            logger.info(String.format("Retained %d / %d sites after filtering on overlap with copy-ratio intervals...",
+                    filteredAllelicCounts.getRecords().size(), unfilteredAllelicCounts.getRecords().size()));
+        }
+
+        if (inputNormalAllelicCountsFile == null) {
+            //filter on homozygosity in case sample
+            logger.info("No matched normal was provided, not running in matched-normal mode...");
+            logger.info("Performing binomial testing and filtering homozygous allelic counts...");
+            hetAllelicCounts = new AllelicCountCollection(
+                    sampleMetadata,
+                    filteredAllelicCounts.getRecords().stream()
+                            .filter(ac -> calculateHomozygousLogRatio(ac, genotypingBaseErrorRate) < genotypingHomozygousLogRatioThreshold)
+                            .collect(Collectors.toList()));
+            final File hetAllelicCountsFile = new File(outputDir, outputPrefix + HET_ALLELIC_COUNTS_FILE_SUFFIX);
+            hetAllelicCounts.write(hetAllelicCountsFile);
+            logger.info(String.format("Retained %d / %d sites after testing for heterozygosity...",
+                    hetAllelicCounts.getRecords().size(), unfilteredAllelicCounts.getRecords().size()));
+            logger.info(String.format("Heterozygous allelic counts written to %s.", hetAllelicCountsFile));
+        } else {
+            //read in matched normal
+            logger.info("Matched normal was provided, running in matched-normal mode...");
+            logger.info("Performing binomial testing and filtering homozygous allelic counts in matched normal...");
+            final AllelicCountCollection unfilteredNormalAllelicCounts = new AllelicCountCollection(inputNormalAllelicCountsFile);
+            if (!unfilteredNormalAllelicCounts.getIntervals().equals(unfilteredAllelicCounts.getIntervals())) {
+                throw new UserException.BadInput("Allelic-count sites in case sample and matched normal do not match. " +
+                        "Run CollectAllelicCounts using the same interval list of sites for both samples.");
+            }
+            final SampleMetadata normalSampleMetadata = unfilteredNormalAllelicCounts.getSampleMetadata();
+
+            //filter on total count in matched normal
+            logger.info(String.format("Filtering allelic counts in matched normal with total count less than %d...", minTotalAlleleCount));
+            final AllelicCountCollection filteredNormalAllelicCounts = new AllelicCountCollection(
```

Comment:
You could extract this and other similar filtering code to a new function.

Reply:
Yup, already filed #3915.